subprojects/core/src/main/java/org/gradle/api/internal/initialization/AbstractClassLoaderScope.java

```diff
@@ -33,11 +37,13 @@
     protected final ClassLoaderScopeIdentifier id;
     protected final ClassLoaderCache classLoaderCache;
     protected final ClassLoaderScopeRegistryListener listener;
+    protected final ClassToInstanceMap<ClassLoaderScopeData> dataMap;
```

Comment:
Is this something that we need to pay attention to when serializing class loaders hierarchy in the context of configuration cache?

Reply:
It doesn't look like it because the data is only used at plugin application time. On the other hand this means every build would be paying the cost for these unused `ClassToInstanceMap` for every node in the tree.

Could this perhaps be solved by externalizing the `ClassLoaderScope` to `ClassLoaderData` associations to a build tree service that would then only keep the relevant data at configuration time?

Reply:
I adopted @bamboo 's changes and removed the `ClassLoaderScopeData` class entirely, as its addition is no longer needed.